src/components/AddPlant.js

```diff
@@ -24,31 +24,37 @@ export const AddPlant = () => {
         'Content-Type': 'application/json',
         Authorization: accessToken
       },
-      // eslint-disable-next-line object-shorthand
-      body: JSON.stringify({ plantname: plantname, species: species })
+      body: JSON.stringify({ plantname, species })
     };
     fetch(API_URL('addplant'), options)
       .then((response) => response.json())
       .then((data) => {
         if (data.success) {
-          dispatch(plants.actions.setPlantname(data.response.plantname));
-          dispatch(plants.actions.setSpecies(data.response.species));
-          dispatch(plants.actions.setPlantId(data.response._id))
-          dispatch(plants.actions.setBirthday(data.response.birthday));
-          dispatch(plants.actions.setLastWatered(data.response.lastWatered));
-          dispatch(plants.actions.setLastSoilChange(data.response.lastSoilChange));
+          // creating a new plant-object with data based on the plant-schema:
+          const newPlant = {
+            _id: data.response._id,
+            plantname,
+            species,
+            birthday: data.response.birthday,
+            lastWatered: data.response.lastWatered,
+            lastSoilChange: data.response.lastSoilChange
+          };
           dispatch(plants.actions.setError(null));
+          // calling the handleAddPlant-function with the newPlant-object
+          // as an argument, comprised of all the data the user passed in themselves
+          handleAddPlant(newPlant);
         } else {
-          dispatch(plants.actions.setPlantname(null));
-          dispatch(plants.actions.setSpecies(null));
-          dispatch(plants.actions.setError(data.response))
+          // dispatching error message if plant object wasn't successfully created
+          dispatch(plants.actions.setError(data.response));
         }
       })
-      .then(
-        setPlantname(''),
-        setSpecies('')
-      )
-  }
+      // emptying the form when the new plant has been added
+      .then(() => {
```

Comment:
I don't know if this one was necessary, it was working beforehand. But looks nicer now

Reply:
Making this into an arrow function makes sure that it doesn't execute immediately, but instead waits until all the previous promises have executed so that all the function runs all its promises in the correct order basically

Reply:
We can also use a .finally?

Reply:
agreed, to make it nicer we need .finally and also some error catching around